contracts/stake/src/contract.rs

```diff
@@ -754,6 +764,82 @@ impl StakingTrait for Staking {
         distribution.distributed_total
     }
 
+    fn update_config(
+        env: Env,
+        lp_token: Option<Address>,
+        min_bond: Option<i128>,
+        min_reward: Option<i128>,
+        manager: Option<Address>,
+        owner: Option<Address>,
+        max_complexity: Option<u32>,
+    ) -> Result<Config, ContractError> {
+        env.storage()
+            .instance()
+            .extend_ttl(INSTANCE_RENEWAL_THRESHOLD, INSTANCE_TARGET_TTL);
+
+        let admin = get_admin_old(&env);
+        admin.require_auth();
+
+        let mut config = get_config(&env);
+
+        if let Some(lp_token) = lp_token {
+            config.lp_token = lp_token;
+        }
+
+        if let Some(min_bond) = min_bond {
+            if min_bond <= 0 {
+                log!(
+                &env,
+                "Stake: initialize: Minimum amount of lp share tokens to bond can not be smaller or equal to 0"
+            );
+                panic_with_error!(&env, ContractError::InvalidMinBond);
+            }
+            config.min_bond = min_bond
+        }
+
+        if let Some(min_reward) = min_reward {
+            if min_reward <= 0 {
+                log!(&env, "Stake: initialize: min_reward must be bigger than 0!");
+                panic_with_error!(&env, ContractError::InvalidMinReward);
+            }
+            config.min_reward = min_reward
+        }
+
+        if let Some(manager) = manager {
+            config.manager = manager;
+        }
+
+        if let Some(owner) = owner {
+            config.owner = owner;
+        }
+
+        if let Some(max_complexity) = max_complexity {
+            if max_complexity == 0 {
+                log!(
+                    &env,
+                    "Stake: initialize: max_complexity must be bigger than 0!"
+                );
+                panic_with_error!(&env, ContractError::InvalidMaxComplexity);
+            }
```

Comment:
Lately I found about `NonZero` math types: https://doc.rust-lang.org/std/num/type.NonZeroU32.html
Let's see if that works with Soroban. If so, we should probably replace that type entirely for that case.

Reply:
oh wow, that's just awesome, and it'd save so much hassle for us. Currently, it's not supported out of the box for Soroban, so we'd have to manually implement a bunch of traits. Let me know if that's something you'd like us do:

```
1. the trait bound `core::num::NonZero<u32>: soroban_sdk::TryFromVal<soroban_sdk::Env, _>` is not satisfied
   the following other types implement trait `soroban_sdk::TryFromVal<E, V>`:
```

Reply:
Wouldn't that be just one simple trait implementation for this case?

Reply:
yes, but that would break the `orhpan rule`. Let me know if you want me to do some wrapper around it `NonZerou32`.